### PR TITLE
Add explicit Java temporal types

### PR DESCRIPTION
**Pull request by EmadAlblueshi**

### CODE REVIEW

File: src/main/java/com/julienviet/pgclient/impl/ArrayTuple.java [hunk 1]
```diff
@@ -85,8 +89,34 @@ public Temporal getTemporal(int pos) {
     return (Temporal) get(pos);
   }
 
+  @Override
+  public LocalDate getLocalDate(int pos) {
+    return (LocalDate) get(pos);
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/ArrayTuple.java]: what should we do if the type is not correct ? can we try to convert it ?

Comment (EmadAlblueshi) [reply src/main/java/com/julienviet/pgclient/impl/ArrayTuple.java]: I think we can check the object type and do the proper conversion if needed

**Comment by vietj** [comment]
can you rebase the PR ? i've squashed a couple of commits in master

**Comment by EmadAlblueshi** [comment]
Yes I will do it now

**Comment by EmadAlblueshi** [comment]
@vietj :+1: Done

**Comment by vietj** [comment]
please rebase without merge, there is a Foo commit that disappeared

**Comment by EmadAlblueshi** [comment]
Thanks

**Pull request merged**